### ISSUE TITLE
decrease memory reads in syscalls realisations

### ISSUE THREAD

**Issue by grishasobol** [gear-tech/gear#1646]
### Problem to Solve

In some syscalls arguments are pointers to the fixed size values in the  program memory, for example: https://github.com/gear-tech/gear/blob/e8af76fd40fb31c3acc7bae3987dd730c5b62eae/gcore/src/msg.rs#L66-L73

This lead to the fact that in syscall realisation we need to access these values from host executor one by one:
https://github.com/gear-tech/gear/blob/e8af76fd40fb31c3acc7bae3987dd730c5b62eae/core-backend/sandbox/src/funcs.rs#L178-L182

Because we use host executor for smart contracts execution to access data from the program memory we need do call host function. Which cause many indirect calls to access target memory buffer and to write the result back in runtime memory.

### First solution
Is to change syscalls signature. For example:
https://github.com/gear-tech/gear/blob/e8af76fd40fb31c3acc7bae3987dd730c5b62eae/gcore/src/msg.rs#L66-L73

In this example we can make special structure for pointers to the static sized values:
```rust
struct Args {
   program_id: ActorId,
   value: u128,
   delay: u32,
}
```
In `gr_send` we can use pointer to the `Args` object:
```rust
        pub fn gr_send(
            args: *const Args,
            data_ptr: *const u8,
            data_len: u32,
            message_id_ptr: *mut u8,
        ) -> SyscallError;
```
And remove `2` memory reads.

### Second solution
Is to make special host function in Sandbox runtime interface which will take slice of `pointers`, `lengths` and `runtime pointers` in order to load all objects from program memory by one host function call. This is more profitable solution because can be used for non-sized objects like payloads. But this solution must be done in substrate part, so we can meet restrictions from parity.

**Comment by breathx** [comment]
> prefer sth like this:
> 
> ```rust
> #[derive(Codec)]
> struct GearSendArguments {
>     ...
> }
> 
> pub fn gr_send(ptr: *const u8) {}
> 
> pub fn send(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
>     // only need to read the pointer then we can decode the args from the type with scale codec.
>     let ptr = pop_i32(args);
>     let args = GearSendArguments::decode(&mut ctx.memory.as_slice()[ptr..]);
> 
>     // ...
> }
> ```

That's an overhead for dynamically sized arguments

**Comment by breathx** [comment]
Fixed in #1847
